src/components/QuickGrid/ColumnPicker/ColumnPicker.tsx

```diff
@@ -0,0 +1,89 @@
+import * as React from 'react';
+import { QuickGrid } from '../QuickGrid';
+
+import * as _ from 'lodash';
+import { GridColumn } from '../QuickGrid.Props';
+import { Checkbox } from '../../Checkbox';
+
+import './ColumnPicker.scss';
+import { Icon } from '../../Icon';
+import { Button } from '../../Button';
+
+export interface ColumnPickerProps {
+    columns: Array<GridColumn>;
+    pickedColumns?: Array<GridColumn>;
+
+    onDismiss?(event?: React.MouseEvent<HTMLElement>);
+    onChanged?(picked: Array<GridColumn>);
+}
+
+export interface ColumnPickerState {
+    isColumnSelected: {[key: string]: boolean};
+}
+
+
+export default class ColumnPicker extends React.Component<ColumnPickerProps, ColumnPickerState> {
+
+    public constructor(props: ColumnPickerProps) {
+        super(props);
+        this.state = {
+            isColumnSelected: this.setVisibleColumns(props.columns, props.pickedColumns)
+        };
+    }
+
+    private setVisibleColumns(columns: Array<GridColumn>, picked?: Array<GridColumn>): {[key: string]: boolean} {
+        let visible: {[key: string]: boolean} = {};
+        columns.forEach(col => visible[col.valueMember] = !!!picked);
+        if (picked) {
```

Comment:
why do we have to iterate through picked columns again? why don't we check it when iterating through all columns (should picked be a subset of columns)? visible[col.valueMember] = !!!picked[col.valueMember]

Reply:
It wouldn't be any more optimized, `picked` is an array of objects, not a dictionary, so I would have to do a lookup for each column.
This way I only do two passes, and not even that because `picked` is a subset of all columns